### PR TITLE
cmake: use project-level variables, not top-level variables

### PR DESCRIPTION
**Pull request by herrecito**
This PR replaces CMAKE_SOURCE_DIR and CMAKE_BINARY_DIR with PROJECT_SOURCE_DIR and PROJECT_BINARY_DIR. This is more correct. It makes possible to add WildMidi as a submodule in a parent project and link it with:

```
add_subdirectory(lib/wildmidi EXCLUDE_FROM_ALL)
target_link_libraries(main PRIVATE libwildmidi_static)
```

It should have no effect on WildMidi itself.

### CODE REVIEW

**Comment by psi29a** [comment]
Cool! Just needs a little love apparently...

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # ########## Project setup ##########
-PROJECT(wildmidi C)
 CMAKE_MINIMUM_REQUIRED(VERSION 2.8.11)
+CMAKE_POLICY(SET CMP0072 NEW)
```
Comment (psi29a) [inline CMakeLists.txt:3]: Instead of setting the policy, could we just bump our cmake requirement?

Comment (sezero) [reply CMakeLists.txt:3]: What do we have to do with FindOpenGL() to need this CMP0072 policy setting? (see: https://cmake.org/cmake/help/v3.11/policy/CMP0072.html)

Comment (psi29a) [reply CMakeLists.txt:3]: We're not an OpenGL consumer, so we shouldn't need to bother with this nor set a policy for it.

Comment (psi29a) [reply CMakeLists.txt:3]: We do not use: `FindOpenGL`

Comment (sezero) [reply CMakeLists.txt:3]: Well, that's exactly the reason I asked, and it seems like the actual author lost interest in his own pull request..

Comment (herrecito) [reply CMakeLists.txt:3]: Sorry for the radio silence! It's been very busy at work, but I'll try to find time this week to get this PR ready!

**Comment by herrecito** [comment]
It should be OK now, though travis failed, but apparently while installing build dependencies on macOS?. I'm pretty sure is not related to this PR.

**Comment by psi29a** [comment]
Yeah, I'll have a look at the MacOS stuff when I have free cycles.

Thanks

**Pull request merged**